transport/controller.cc

```diff
@@ -65,6 +66,124 @@ future<> controller::start_server() {
     return do_start_server().finally([this] { _ops_sem.signal(); });
 }
 
+// Wrapper for cql_server::listen to invoke it on all shards and log address and listen options.
+static future<> listen(sharded<cql_server>& cserver, socket_address addr, std::shared_ptr<seastar::tls::credentials_builder> creds, bool is_shard_aware, bool keepalive, std::optional<file_permissions> unix_domain_socket_permissions) {
```

Comment:
Maybe better to assign `unix_domain_socket_permissions` a default value instead of making it optional, that way you won't have to pass an extra `std::nullopt` if you don't intend to specify permissions.

Reply:
not sure if this makes sense because afacs seastar part of this patch is either doing some calls or not (use defaults) so later it would require extra check `if unix_domain_socket_permissions != default` 

Documennthing this in some way in the comment in generic_server.hh could be nice though

Reply:
There's a difference between setting permissions to a default value, and not changing permissions.